src/page/Profile/Profile.jsx

```diff
@@ -1,63 +1,110 @@
-import React from "react";
-import { PostCard } from "../../components/componentExport";
+import React, { useEffect, useState } from "react";
+import {
+  PostCard,
+  Loader,
+  ProfileModal,
+} from "../../components/componentExport";
 import { AiOutlineLogout } from "react-icons/ai";
 import { logoutUser } from "../../features/authSlice";
-import { useDispatch } from "react-redux";
-import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams } from "react-router-dom";
+import { getAllUsers } from "../../features/usersSlice";
 
 export const Profile = () => {
+  const [isEditProfile, setIsEditProfile] = useState(false);
+
   const dispatch = useDispatch();
-  const navigate = useNavigate();
+  const { userName } = useParams();
+  const {
+    auth: { user },
+    allUsers: { users, isLoading },
+  } = useSelector((state) => state);
+
+  const currentUser = users?.find((user) => user.username === userName);
 
-  const logoutHandler = () => {
-    dispatch(logoutUser());
-    navigate("/");
-  };
+  useEffect(() => {
+    dispatch(getAllUsers());
+  }, []);
```

Comment:
I see you are fetching all the users on the postPage as well, any particular reason?

Reply:
For user suggestions I need all users, that's why.